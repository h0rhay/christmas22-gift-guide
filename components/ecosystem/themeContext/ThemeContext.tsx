import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
