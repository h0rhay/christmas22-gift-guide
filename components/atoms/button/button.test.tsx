import { render } from "@testing-library/react";
import Button, { ButtonVariationType } from ".";
import { ThemeProvider } from "styled-components";
import Theme from "styles/theme";
import ThemeContext from "components/ecosystem/themeContext/ThemeContext";

const renderComponent = (buttonVariation?: ButtonVariationType) =>
  render(
    <ThemeProvider theme={Theme}>
      <ThemeContext colorGroup="hero">
        <Button type="button" variation={buttonVariation}>
          <a className="button__a" href="">
            foo
          </a>
        </Button>
      </ThemeContext>
    </ThemeProvider>
  );

describe("Button", () => {
  it("should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  // @TODO It would be great to solve the problem with screen sizes on styled-components to create additional tests
  it("Sponsor Button in Mobile View should have background sfYellow font color as sfYellow", () => {
    const { container, getByRole } = renderComponent("sponsor");
    const buttonEl = getByRole("button");
    const aEl = container.getElementsByClassName("button__a")[0];

    expect(buttonEl).toHaveStyle("background: rgba(255,226,85,1)");
    expect(aEl).toHaveStyle("color: rgba(254, 251, 240, 1)");
  });

  it("Sponsor-reverse Button as default (mobile) should have background none font color as sfYellow", () => {
    const { container, getByRole } = renderComponent("sponsor-reverse");
    const buttonEl = getByRole("button");
    const aEl = container.getElementsByClassName("button__a")[0];

    expect(buttonEl).toHaveStyle("background: none");
    expect(aEl).toHaveStyle("color: rgba(254, 251, 240, 1)");
  });
});
