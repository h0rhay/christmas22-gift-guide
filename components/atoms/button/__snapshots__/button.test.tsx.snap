// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <button
      class="button__ButtonStyles-sc-lri851-0 guzMzG"
      type="button"
    >
      <a
        class="button__a"
        href=""
      >
        foo
      </a>
    </button>
  </div>
</div>
`;
